### DISCUSSION TITLE
How to access MRTK 3 profile that is present in project settings via code at runtime

### DISCUSSION THREAD

**Discussion by Ram1508** [microsoft/MixedRealityToolkit-Unity#11571] [Q&A]
@AMollis 
I need to change the profile based on the target device  to provide related service .
![Screenshot (30)](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/131968892/ce9f673f-56a0-4fde-b703-f773df7d0011)

**Comment by AMollis** [comment]
Hi @Ram1508 , thank you for your question.  Can you clarify your scenario and what you are trying to do?  You can't access the MRTK3 configuration at runtime, but you can access the corresponding sub-systems.  Are you trying to access a particular sub-system at runtime?

**Comment by Ram1508** [comment]
Hi @AMollis, 
At the start of the project we manually need to assign Mrtk3 profile. 
I am trying to assign different profile for different platform in Editor script.


**Scenario**

I am trying to build my project  on hololens 2 and quest 2, but there are certain subsystems that needed on one and not on another including the custom made subsystem. So instead of maintaining  a same profile and enable/disable the subsystems, I thought of maintaining different pre made profile with required subsystems being enabled so that one just need to assign the profile based upon the target device.

**Steps I tried**

1. MRTKProfile.cs :
![MRTKProfileInstance](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/131968892/901958d6-702a-42f9-aab8-f37639a4d569)


Using this I was able to change the instance of profile at runtime. But I also wanted it to reflect on the editor screen.

2.MRTKSettingsEditor.cs

![Screenshot (34)](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/131968892/20e1fadd-d3d7-4219-adef-b330b93debd8)

OnInspectorGUI() function checks for the assignment of profile and assign default profile if not.
I need to replicate that but certain properties are encapsulated.

Is there anyotherway to achieve this?

**Reply by AMollis** [reply]
@Ram1508 the UI already supports configuring MRTK differently based on platform type.  You should be able to select the Android tab and configure for Quest, and select the Windows icon to configure for HoloLens 2.  Is this not working for your scenario?

![image](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/36461279/e50b48f8-dfbd-4cb7-b703-15c3b68f73ae)

**Comment by Ram1508** [comment]
@AMollis  thanks for the reply,
I was able to change the profile via UI, but I am trying to provide my project as an sdk, so i was trying to pre-assign profile for different platform via editor code instead of insisting the developers to do so.
By providing these profiles, developers using the SDK can seamlessly integrate it into various platforms, such as iOS, Android, web, or desktop applications.
